src/main/java/com/ryanharter/auto/value/parcel/Parcelables.java

```diff
@@ -0,0 +1,221 @@
+  private static final TypeName MAP = ClassName.get("java.util", "Map");
+  private static final TypeName LIST = ClassName.get("java.util", "List");
+  private static final TypeName BOOLEANARRAY = ArrayTypeName.of(boolean.class);
+  private static final TypeName BYTEARRAY = ArrayTypeName.of(byte.class);
+  private static final TypeName INTARRAY = ArrayTypeName.of(int.class);
+  private static final TypeName LONGARRAY = ArrayTypeName.of(long.class);
+  private static final TypeName STRINGARRAY = ArrayTypeName.of(String.class);
+  private static final TypeName SPARSEARRAY = ClassName.get("android.util", "SparseArray");
+  private static final TypeName SPARSEBOOLEANARRAY = ClassName.get("android.util", "SparseBooleanArray");
+  private static final TypeName BUNDLE = ClassName.get("android.os", "Bundle");
+  private static final TypeName PARCELABLE = ClassName.get("android.os", "Parcelable");
+  private static final TypeName PARCELABLEARRAY = ArrayTypeName.of(PARCELABLE);
+  private static final TypeName CHARSEQUENCE = ClassName.get("java.lang", "CharSequence");
+  private static final TypeName CHARSEQUENCEARRAY = ArrayTypeName.of(CHARSEQUENCE);
+  private static final TypeName IBINDER = ClassName.get("android.os", "IBinder");
+  private static final TypeName OBJECTARRAY = ArrayTypeName.of(TypeName.OBJECT);
+  private static final TypeName SERIALIZABLE = ClassName.get("java.io", "Serializable");
+  private static final TypeName PERSISTABLEBUNDLE = ClassName.get("android.os", "PersistableBundle");
+  private static final TypeName SIZE = ClassName.get("android.util", "Size");
+  private static final TypeName SIZEF = ClassName.get("android.util", "SizeF");
+
+  private static final List<TypeName> VALID_TYPES = Arrays.asList(STRING, MAP, LIST, BOOLEANARRAY,
+      BYTEARRAY, INTARRAY, LONGARRAY, STRINGARRAY, SPARSEARRAY, SPARSEBOOLEANARRAY, BUNDLE,
+      PARCELABLE, PARCELABLEARRAY, CHARSEQUENCE, CHARSEQUENCEARRAY, IBINDER, OBJECTARRAY,
+      SERIALIZABLE, PERSISTABLEBUNDLE, SIZE, SIZEF);
+
+  public static boolean isValidType(TypeName typeName) {
+    return typeName.isPrimitive() || VALID_TYPES.contains(typeName);
+  }
+
+  public static boolean isValidType(Types types, TypeElement type) {
+    return getParcelableType(types, type) != null;
+  }
+
+  public static TypeName getParcelableType(Types types, TypeElement type) {
+    TypeMirror typeMirror = type.asType();
+    while (typeMirror.getKind() != TypeKind.NONE) {
+
+      // first, check if the class is valid.
+      TypeName typeName = TypeName.get(typeMirror);
+      if (typeName instanceof ParameterizedTypeName) {
+        typeName = ((ParameterizedTypeName) typeName).rawType;
+      }
+      if (typeName.isPrimitive() || VALID_TYPES.contains(typeName)) {
+        return typeName;
+      }
+
+      // then check if it implements valid interfaces
+      List<? extends TypeMirror> interfaces = type.getInterfaces();
+      if (!interfaces.isEmpty()) {
```

Comment:
Don't really need this check which lets you inline the `getInterfaces()` call into the `for`